Dockerfile

```diff
@@ -1,15 +1,37 @@
-FROM golang:alpine
+ARG GO_VERSION=1.17
+ARG ALPINE_VERSION=3.15
+
+# Build stage
+FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as build
 
 WORKDIR /app
 
-COPY go.mod .
-COPY go.sum .
+COPY go.mod go.sum .
+
 RUN go mod download
 
-COPY . .
+COPY cmd cmd
+
+COPY server server
+
+RUN CGO_ENABLED=0 go build \
+    -ldflags="-extldflags=-static" \
```

Comment:
If it turns out we don't want multi-stage builds (see below comment), `CGO_ENABLED=0` and `-ldflags="-extldflags=-static"` are unnecessary